io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/XLSInstanceReader.java

```diff
@@ -83,83 +74,121 @@ public boolean isCancelable() {
+			ReaderSettings settings = ReaderSettings.load(this);
+			Collection<? extends SheetInfo> sheets = settings.getSheetsToRead();
+			if (sheets.isEmpty()) {
+				reporter.info("No sheets to load");
+				reporter.setSuccess(true);
+				return reporter;
+			}
 
-		// first sheet as default
-		sheetNum = getParameter(InstanceTableIOConstants.SHEET_INDEX).as(int.class, 0);
-
-		instances = new DefaultInstanceCollection(new ArrayList<Instance>());
+			// instance collection where all instances are collected
+			instances = new DefaultInstanceCollection(new ArrayList<Instance>());
+
+			boolean failed = false;
+			for (SheetInfo sheet : sheets) {
+				try {
+					loadSheet(sheet, reporter);
+				} catch (Exception e) {
+					failed = true;
+					reporter.error("Reading the excel sheet {0} at index {1} failed",
+							sheet.getName(), sheet.getIndex(), e);
+				}
+			}
 
-		try {
-			// analyze the excel sheet to get all information
-			analyser = new AnalyseXLSSchemaTable(getSource().getLocation(), sheetNum);
+			reporter.setSuccess(!failed);
 		} catch (Exception e) {
-			reporter.error(new IOMessageImpl("Reading the excel sheet has failed", e));
-			return reporter;
+			reporter.error("Reading the excel file failed", e);
+		} finally {
+			progress.end();
 		}
+		return reporter;
+	}
 
-		// get type definition of the schema
-		type = getSourceSchema().getType(
-				QName.valueOf(getParameter(CommonSchemaConstants.PARAM_TYPENAME).as(String.class)));
-
-		// get property definition
-		propAr = type.getChildren().toArray(new PropertyDefinition[type.getChildren().size()]);
-		Collection<List<String>> rows = analyser.getRows();
+	private void loadSheet(SheetInfo sheet, IOReporter reporter) throws Exception {
+		AnalyseXLSSchemaTable analyser = new AnalyseXLSSchemaTable(getSource(),
+				ReaderSettings.isXlsxContentType(getContentType()), sheet.getIndex());
 
-		int skipNlines = 0;
-		Boolean skipType = getParameter(CommonSchemaConstants.PARAM_SKIP_N_LINES).as(Boolean.class);
+		// get type definition of the schema
+		QName typeName = sheet.getSettings().getTypeName();
+		TypeDefinition type = null;
+		if (typeName != null) {
+			type = getSourceSchema().getType(typeName);
+		}
+		if (type == null) {
+			// look for match based on sheet name
+			String localName;
+			QName qname = null;
+			try {
+				qname = QName.valueOf(sheet.getName());
+				localName = qname.getLocalPart();
+			} catch (Exception e) {
+				localName = sheet.getName();
+			}
+			final String flName = localName;
 
-		if (skipType == null) {
+			// exact name
+			if (qname != null) {
+				type = getSourceSchema().getType(qname);
+			}
 
-			skipNlines = getParameter(CommonSchemaConstants.PARAM_SKIP_N_LINES).as(Integer.class);
+			if (type == null) {
+				// try display name (since this is used when writing to Excel)
+				getSourceSchema().getMappingRelevantTypes().stream()
+						.filter(t -> flName.equals(t.getDisplayName())).findFirst().orElse(null);
+			}
 
+			if (type == null) {
```

Comment:
`type` is not set in the above `filter()` operation it would still be `null` here if it was `null` above. Also is it intended to leave the result of this `filter()` operation?